---
title: "**Variance All the Way Down:** Exploring the Impact of RNA-Seq Pipeline Choices on Differential Expression Variance"
author: "Hunter Schuler and Art Tay"
format:
  pdf:
     documentclass: article
     papersize: letter
     geometry:
         margin=1in
     include-in-header: header.tex
bibliography: references.bib
csl: american-statistical-association.csl
---

```{r setup, include=FALSE}
##Setup code
knitr::opts_chunk$set(warning = FALSE, message = FALSE)

# Libraries
library(tidyverse)
library(tidymodels)
library(kableExtra)

tidymodels_prefer()
```

# Analysis

```{r}
# Meta Data
sample_names <- c(
    "gene",
    "SRR31476642",
    "SRR31476643",
    "SRR31476644",
    "SRR31476645",
    "SRR31476646",
    "SRR31476647",
    "SRR31476648",
    "SRR31476649",
    "SRR31476650"
)

treatments <- c(
    "DMSO",
    "DMSO",
    "DMSO",
    "DMSO",
    "EPZ015666",
    "EPZ015666",
    "EPZ015666",
    "DMSO",
    "DMSO"
)

factors <- c("aligner", "trim_poly_g", "trim_poly_x", "norm_method")
```

## Counts
```{r}
# Read in data. 
count_sd_df <- read.csv("./data/gen_samples/count_sd_df.csv")
count_sd_df <- count_sd_df |> 
    mutate(across(any_of(factors), ~ as.factor(.)))

# Classic LM 
lm_fit <- count_sd_df |> 
    select(-c(runtime_sec, gene_overlap_percent)) |> 
    (\(x) glm(count_sd ~ (.)^2, family = gaussian(), data = x))()
summary(lm_fit)

# Log-normal GLM 
glm_log_fit <- count_sd_df |> 
    select(-c(runtime_sec, gene_overlap_percent)) |> 
    (\(x) glm(count_sd ~ (.)^2, family = gaussian(link = "log"), data = x))()
summary(glm_log_fit)

# Quasi GLM
quasi_fit <- count_sd_df |>
    select(-c(runtime_sec, gene_overlap_percent)) |> 
    (\(x) glm(count_sd ~ (.)^2, family = quasi(), data = x))()
summary(quasi_fit)
```

## P-Values
```{r}
# Read in data. 
DE_sd_df <- read.csv("./data/gen_samples/DE_sd_df.csv")
DE_sd_df <- DE_sd_df |> 
    mutate(across(any_of(factors), ~ as.factor(.)))

# Classic LM 
lm_fit <- DE_sd_df |> 
    select(-c(runtime_sec, gene_overlap_percent, effect_size_sd)) |> 
    (\(x) glm(p_value_sd ~ (.)^2, family = gaussian(), data = x))()
summary(lm_fit)

# Log-normal GLM 
glm_log_fit <- DE_sd_df |> 
    select(-c(runtime_sec, gene_overlap_percent, effect_size_sd)) |> 
    (\(x) glm(p_value_sd ~ (.)^2, family = gaussian(link = "log"), data = x))()
summary(glm_log_fit)

# Quasi GLM
quasi_fit <- DE_sd_df |>
    select(-c(runtime_sec, gene_overlap_percent, effect_size_sd)) |> 
    (\(x) glm(p_value_sd ~ (.)^2, family = quasi(), data = x))()
summary(quasi_fit)
```

## Effect Size
```{r}
# Read in data. 
DE_sd_df <- read.csv("./data/gen_samples/DE_sd_df.csv")
DE_sd_df <- DE_sd_df |> 
    mutate(across(any_of(factors), ~ as.factor(.)))

# Classic LM 
lm_fit <- DE_sd_df |> 
    select(-c(runtime_sec, gene_overlap_percent, p_value_sd)) |> 
    (\(x) glm(effect_size_sd ~ (.)^2, family = gaussian(), data = x))()
summary(lm_fit)

# Log-normal GLM 
glm_log_fit <- DE_sd_df |> 
    select(-c(runtime_sec, gene_overlap_percent, p_value_sd)) |> 
    (\(x) glm(effect_size_sd ~ (.)^2, family = gaussian(link = "log"), data = x))()
summary(glm_log_fit)

# Quasi GLM
quasi_fit <- DE_sd_df |>
    select(-c(runtime_sec, gene_overlap_percent, p_value_sd)) |> 
    (\(x) glm(effect_size_sd ~ (.)^2, family = quasi(), data = x))()
summary(quasi_fit)
```